### ISSUE TITLE
sequelize.Utils.toDefaultValue(Sequelize.UUIDV4) returns empty object

### ISSUE THREAD

**Issue by tiblu** [sequelize/sequelize#3324]
**Is**
``sequelize.Utils.toDefaultValue(Sequelize.UUIDV4)`` returns empty object ({})

**Expected**
``sequelize.Utils.toDefaultValue(Sequelize.UUIDV4)`` to return valid UUIDV4

Sequelize version: **2.0.4**
Was OK in Sequelize **2.0.0-RC3**

## Test ##

**Code:**
https://bitbucket.org/tiblu/experiments/src/3e382d9f599860278cc341f6888a0cd29c393ab3/sequelize/models/defaultValue.js?at=master
```
'use strict';

var Sequelize = require('sequelize');
var lodash = require('lodash');


var sequelize = new Sequelize('postgres://experiments:experiments@localhost:5433/experiments',
    {
        "logging": console.log,
        "sync": {
            "force": true,
            "logging": console.log
        },
        "define": {
            "underscored": false,
            "freezeTableName": false,
            "syncOnAssociation": false,
            "charset": "utf8",
            "collate": "utf8_general_ci",
            "timestamps": true,
            "paranoid": true
        },
        "omitNull": true
    }
);

sequelize.sync().then(function () {
    console.log('---------------- Sync success ----------------');
    var expectedToBeUUID = sequelize.Utils.toDefaultValue(Sequelize.UUIDV4);
    console.log('sequelize.Utils.toDefaultValue(Sequelize.UUIDV4) = ', expectedToBeUUID, typeof expectedToBeUUID);
    console.log('Sequelize.UUIDV4 lodash.isFunction = ', typeof sequelize.Utils.toDefaultValue(Sequelize.UUIDV4));
    console.log('sequelize.Utils.toDefaultValue is = ', sequelize.Utils.toDefaultValue); // Lets see what function body is...
    console.log('Sequelize.UUIDV4 is = ', Sequelize.UUIDV4); // Lets see what Sequelize.UUIDV4 is ...
    console.log('Sequelize.UUIDV4 lodash.isFunction = ', lodash.isFunction(Sequelize.UUIDV4));
});
```

**Output:**
```
---------------- Sync success ----------------
sequelize.Utils.toDefaultValue(Sequelize.UUIDV4) =  {} object
Sequelize.UUIDV4 lodash.isFunction =  object
sequelize.Utils.toDefaultValue is =  function (value) {
    if (lodash.isFunction(value)) {
      return value();
    } else if (value instanceof DataTypes.UUIDV1) {
      return uuid.v1();
    } else if (value instanceof DataTypes.UUIDV4) {
      return uuid.v4();
    } else if (value instanceof DataTypes.NOW) {
      return Utils.now();
    } else {
      return value;
    }
  }
Sequelize.UUIDV4 is =  function () {
  if (!(this instanceof UUIDV4)) return new UUIDV4();
  ABSTRACT.apply(this, arguments);
}
Sequelize.UUIDV4 lodash.isFunction =  true
```

**Comment by mickhansen** [comment]
How about `sequelize.Utils.toDefaultValue(Sequelize.UUIDV4())`?

**Comment by tiblu** [comment]
@mickhansen Thanks, that works
´´sequelize.Utils.toDefaultValue(Sequelize.UUIDV4()) =  9da6749b-de58-457b-96a8-5dc6e791660f´´

The behaviour has changed from 2.0.0-RC3 to 2.0.4
What do you think, is it an accident or meant to be so?

**Comment by mickhansen** [comment]
@tiblu no it should still work, if you use `defaultValue: UUIDV4` it should be calling `new` automatically.

**Comment by BridgeAR** [comment]
I can't reproduce the error. It seems to be already fixed in 2.1.3?
